File: Android/app/src/main/java/com/arcinteractive/spaces/data/model/MessageType.kt
package com.arcinteractive.spaces.data.model

enum class MessageType(
    val displayName: String,
    val placeholderIconName: String
) {
    Text("Text", "text"),
    Image("Image", "image"),
    Video("Video", "video"),
    Meme("Meme", "meme"),
    Gif("Gif", "gif"),
    Screenshot("Screenshot", "screenshot"),
    File("File", "file")

    ;

    val isPhotosModuleSupported: Boolean
        get() = when (this) {
            Image, Video -> true
            Text, Meme, Gif, Screenshot, File -> false
        }
}

File: iOS/Spaces V1.5/Models/MessageType.swift
import Foundation

enum MessageType: String, CaseIterable, Identifiable, Hashable {
    case text
    case image
    case video
    case meme
    case gif
    case screenshot
    case file

    var id: String { rawValue }

    var systemImageName: String {
        switch self {
        case .text: return "text.bubble"
        case .image: return "photo"
        case .video: return "play.rectangle.fill"
        case .meme: return "face.smiling.inverse"
        case .gif: return "sparkles.tv"
        case .screenshot: return "rectangle.on.rectangle"
        case .file: return "doc.fill"
        }
    }

    var displayName: String {
        rawValue.capitalized
    }

    var isPhotosModuleSupported: Bool {
        switch self {
        case .image, .video:
            return true
        case .text, .meme, .gif, .screenshot, .file:
            return false
        }
    }
}
